thunder/executors/apex_fused_rms_norm_impl.py

```diff
@@ -0,0 +1,123 @@
+    input: TensorLike, weight: TensorLike, normalized_shape: Sequence[int], eps: float, memory_efficient: bool
+):
+    ctx = Context()
+    output = FusedRMSNormAffineMixedDtypesFunction.forward(ctx, input, weight, normalized_shape, eps, memory_efficient)
+    return output, ctx.pop_saved_tensors(), ctx
+
+
+fused_rms_norm_fwd = apex_ex.register_operator("fused_rms_norm_fwd", meta=meta_impl_fn, fn=fused_rms_norm_impl)
+
+
+def fused_rms_norm_backward_meta(saved_tensors: Sequence[torch.Tensor], ctx: Context, g: TensorLike):
+    # saved_tensors[0] - input or output
+    # saved_tensors[1] - weight
+    return TensorProxy(like=saved_tensors[0]), TensorProxy(like=saved_tensors[1])
+
+
+def fused_rms_norm_backward_impl(saved_tensors: Sequence[torch.Tensor], ctx: Context, g: TensorLike):
+    with set_saved_tensors(ctx, saved_tensors):
+        return FusedRMSNormAffineMixedDtypesFunction.backward(ctx, g)[:2]
+
+
+fused_rms_norm_backward = apex_ex.register_operator(
+    "fused_rms_norm_backward", meta=fused_rms_norm_backward_meta, fn=fused_rms_norm_backward_impl
+)
+
+
+def fused_rms_norm_grad_rule(
+    input: TensorLike, weight: TensorLike, normalized_shape: Sequence[int], eps: float, memory_efficient: bool
+):
+    output, saved_tensors, saved_meta = fused_rms_norm_fwd(input, weight, normalized_shape, eps, memory_efficient)
+    g = get_grad(output)
+    grad_input, grad_weight = fused_rms_norm_backward(saved_tensors, saved_meta, g)
+    put_grads((input, weight), (grad_input, grad_weight))
+    return output
+
+
+def execution_tfms(
+    input: TensorLike, weight: TensorLike, normalized_shape: Sequence[int], eps: float, memory_efficient: bool
+):
+    output, _, _ = fused_rms_norm_fwd(input, weight, normalized_shape, eps, memory_efficient)
+    return output
+
+
+def _fused_rms_norm_checker(
+    input: TensorLike, weight: TensorLike, normalized_shape: Sequence[int], eps: float, memory_efficient: bool
+):
+    use_apex_fused_rms_norm = get_compile_option(
+        "use_apex_fused_rms_norm", "Whether to enable `fused_rms_norm` from `apex_ex`. Defaults to `True`."
+    )
+    # We explicitly check for `False` as if the value is unspecified by user, `get_compile_option` returns `None` and `not None` is equal to True.
+    if use_apex_fused_rms_norm == False:  # User explicitly disabled this.
+        return False
+
+    # use_apex_fused_rms_norm is `None` or `True`.
+    return True
+
+
+apex_ex.register_implementation(
+    fused_rms_norm,
+    execution_transform=execution_tfms,
+    grad_transform=fused_rms_norm_grad_rule,
+    checker=_fused_rms_norm_checker,
+)
+apex_ex.register_implementation(fused_rms_norm_backward, fused_rms_norm_backward)
+
+
+# Register the lookaside.
+def maybe_register_apex_fused_rms_norm_lookaside() -> None:
+    if APEX_FUSED_NORMS_AVAILABLE:
+
+        @thunder.core.jit_ext.register_general_jit_lookaside(FusedRMSNormAffineMixedDtypesFunction.forward)
+        @thunder.core.jit_ext.interpreter_needs_wrap
```

Comment:
Could this be handled by `replaces=FusedRMSNormAffineMixedDtypesFunction.forward` on the register_operator instead?

Reply:
Seems good to me.